js/app.js

```diff
@@ -1,5 +1,57 @@
-angular.module('GiphyApp', [])
+var app = angular.module('GiphyApp', ['infinite-scroll']);
+
+app.controller('GiphySearch', ['$scope', '$http', function($scope, $http) {
+  $scope.userSearch = '';
+
+  $scope.searchForGifs = function() {
+    $http.get('http://api.giphy.com/v1/gifs/search?', {
+      params: {
+        q: $scope.userSearch,
+        api_key: "dc6zaTOxFJmzC"
+      }
+    })
+    .then(function success(response) {
+      console.log(response);
+      $scope.gifs = response.data.data
+
+    }, function error(response) {
+      console.log('we got an error response: ');
+    });
+  };
+
+  $scope.$watch('userSearch', function(newValue, oldValue) {
+    $http.get('http://api.giphy.com/v1/gifs/search?', {
+      params: {
+        q: $scope.userSearch,
+        limit: 5,
+        api_key: "dc6zaTOxFJmzC"
+      }
+    })
+    .then(function success(response) {
+      console.log(response);
+      $scope.gifs = response.data.data
+
+    }, function error(response) {
+      console.log('we got an error response: ');
+    });
+  });
+
+  $scope.searchMore = function() {
+    $http.get('http://api.giphy.com/v1/gifs/search?', {
+      params: {
+        q: $scope.userSearch,
+        offset: 5,
+        api_key: "dc6zaTOxFJmzC"
+      }
+    })
+    .then(function success(response) {
+      console.log(response);
+      $scope.gifs = response.data.data
+
+    }, function error(response) {
+      console.log('we got an error response: ');
+    });
+  }
```

Comment:
Here is some code from my deliverable to handle the infinite scrolling:

    $scope.search = function () {
        if ($scope.searchTerm != $scope.oldSearchTerm) {$scope.giphys = [];};
        var req = {
            url: "http://api.giphy.com/v1/gifs/search",
            method: 'GET',
            params: {
                api_key: $scope.api,
                offset: $scope.offset,
                q: $scope.searchTerm,
            }
        }
        $http(req).then(function success(res) {
            $scope.giphys = $scope.giphys.concat(res.data.data);
        }, function error (res) {
            console.log(res);
        });
    };

    $scope.myPagingFunction = function (){
        $scope.oldSearchTerm = $scope.searchTerm;
        $scope.offset = 25 * $scope.count;
        $scope.search();
        $scope.count++;
    };